File: UrlShorter/Server/Models/Url.cs
using Microsoft.AspNetCore.Identity;

namespace Server.Models
{
    public class Url
    {
        public int Id { get; set; }
        public required string FullUrl { get; set; }
        public required string ShortUrl { get; set; }
        
        //CreatedBy
        public required string UserId { get; set; }
        public required DateTime CreatedDate { get; set; }

        public IdentityUser User { get; set; }
    }
}

File: UrlShorter/Client/src/models/url/Url.ts
import { UserResponse } from "../UserResponse";

export interface Url{
    id: number,
    fullUrl: string,
    shortUrl: string,
    userId: string,
    createdDate: string,
}

export interface UrlResponse {
    id: number,
    fullUrl: string,
    shortUrl: string,
    userId: string,
    createdDate: string,
    user: UserResponse
}